resources/assets/js/photoAlbum.js

```diff
@@ -0,0 +1,315 @@
+var items = [];
+var count = 0;
+try {
+    var currentPhotoAlbum = photoAlbum.id;
+    for (var photo in photos.data) {
+        items.push({ src: photos.data[photo].link, w: photos.data[photo].width, h: photos.data[photo].height });
+    }
+    $("p#albumDesciption").text($("p#albumDesciption").text().replace(/\n/g, "<br>"));
+}
+catch(err) {
+    $("p#albumDesciption").text($("p#albumDesciption").text().replace(/\n/g, "<br>"));
+}
+
+function openGallery(index) {
+    var pswpElement = document.querySelectorAll('.pswp')[0];
+    var options = {
+        index: parseInt(index)
+    };
+    var gallery = new PhotoSwipe(pswpElement, PhotoSwipeUI_Default, items, options);
+    gallery.init();
+}
+
+//button spinner
+function ChangebuttonState(text){
+    var button = $('button#submit');
+    var loadingText = '<i class="icon ion-refreshing"></i> ' + text;
+    if (button.html() !== loadingText) {
+        button.data('original-text', button.html());
+        button.html(loadingText);
+        button.attr("disabled", true);
+    }
+}
+
+function addAlbum(thumbnailResults, photoResults, albumName, albumDescription, captureDate) {
+    var formData = new FormData();    
+    formData.append("title", albumName);
+    formData.append("description", albumDescription);
+    formData.append("captureDate", captureDate);
+    if(thumbnailResults.length< 5){
+        index= thumbnailResults.length
+    } else index = 5;
+    for(var i=0; i < index; i++){
+        formData.append('thumbnails[]',thumbnailResults[i]);
+        formData.append('photos[]', photoResults[i]);
+    }
+    
+    var type = "POST";
+    formData.append("_token", window.Laravel.csrfToken);
+
+    $.ajax({
+        data: formData,
+        type: type,
+        contentType: false,
+        processData: false,
+        success: function (result) {
+            currentPhotoAlbum = "photoalbums/" + result;
+            if(thumbnailResults.length> 5){
+                addPhotoToAlbum(thumbnailResults, photoResults, index);
+            } else{
+                location.reload();
+            }
+        },
+        error: function (request, error) {
+            console.log(arguments);
+            alert(" Can't do because: " + error);
+        }
+    });
+}
+
+function addPhotoToAlbum(thumbnailResults, photoResults, index){
+    var i = index;
+    var j = 0;
+    if((thumbnailResults.length - index) <= 5){
```

Comment:
Iets van comments zouden gewaardeerd worden ;) Bv wat i en j hier precies betekenen. Geldt voor heel dit bestand eigenlijk

Reply:
fixed in commit 2bf769d